web-common/src/components/button-toggle-group/SubButton.svelte

```diff
@@ -0,0 +1,72 @@
+<script lang="ts">
+  import { getContext } from "svelte";
+  import {
+    ButtonGroupContext,
+    buttonGroupContext,
+  } from "./ButtonToggleGroup.svelte";
+
+  import Tooltip from "../tooltip/Tooltip.svelte";
+  import TooltipContent from "../tooltip/TooltipContent.svelte";
+
+  export let key: number | string;
+  export let tootips: {
+    selected?: string;
+    unselected?: string;
+    disabled?: string;
+  } = undefined;
+
+  const {
+    registerSubButton,
+    toggleSubButton,
+    selectedKey,
+    firstKey,
+    lastKey,
+    disabledKeys,
+  }: ButtonGroupContext = getContext(buttonGroupContext);
+
+  registerSubButton(key);
+
+  $: isDisabled = disabledKeys.includes(key);
+  $: isSelected = $selectedKey === key;
+
+  const baseStyles = `shrink flex flex-row items-center px-1 py-1
+  transition-transform duration-100`;
+
+  $: textStyle = isDisabled
+    ? "text-gray-400"
+    : "text-gray-700 hover:text-gray-900 ";
+
+  $: bgStyle = isDisabled
+    ? "bg-white"
+    : isSelected
+    ? "bg-gray-100 hover:bg-gray-200 "
+    : "bg-white hover:bg-gray-50 ";
+
+  // This is needed to make sure that the left and right most child
+  // elements don't break out of the border drawn by the parent element
+  $: isFirst = key === $firstKey;
+  $: isLast = key === $lastKey;
```

Comment:
See my comments about making these manually maintained stores, as opposed to deriving them from the single source of truth which is the `subButtons` array. This would prevent potential bugs like we have in `onDestroy()`, where it seems we forgot to update firstKey/lastKey stores

Perhaps `subButtons` is what needs to be a writable store, with firstKey as derived from that.

Or alternatively, just expose a function called `getKeyIndex()` or `isFirstKey()` from the ButtonGroup context